_pytest/recwarn.py

```diff
@@ -13,83 +17,171 @@ def pytest_funcarg__recwarn(request):
     See http://docs.python.org/library/warnings.html for information
     on warning categories.
     """
-    if sys.version_info >= (2,7):
-        oldfilters = warnings.filters[:]
-        warnings.simplefilter('default')
-        def reset_filters():
-            warnings.filters[:] = oldfilters
-        request.addfinalizer(reset_filters)
     wrec = WarningsRecorder()
```

Comment:
Minor: this fits nicely as a yield_fixture:

```python
@pytest.yield_fixture
def recwarn():
    wrec = WarningsRecorder()
    with wrec:
        warnings.simplefilter('default')
        yield wrec
```

 (should've seen it before, sorry! :sweat_smile:)